src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,36 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MIN_REGISTRATION_AGE = 18;
+    public static final int MAX_REGISTRATION_AGE = 100;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MAX_LOGIN_LENGTH = 20;
+    private static final int MAX_PASSWORD_LENGTH = 20;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RuntimeException {
+        if (user == null) {
+            throw new RuntimeException("User cannot be null");
+        }
+        if (user.getAge() < MIN_REGISTRATION_AGE || user.getAge() > MAX_REGISTRATION_AGE
+                || user.getAge() == null) {
+            throw new RuntimeException("User`s age cannot be less than "
+                    + MIN_REGISTRATION_AGE + " and more than " + MAX_REGISTRATION_AGE);
+        }
+        if (user.getLogin().length() == 0 || user.getLogin().length() > MAX_LOGIN_LENGTH
+                || user.getLogin() == null) {
+            throw new RuntimeException("User`s login cannot be empty");
```

Comment:
it would be better to use `.isBlank()` and check `user.getLogin() == null` before string value